### DISCUSSION TITLE
Support functional fit in pyhf

### DISCUSSION THREAD

**Discussion by prbbing** [pyhf/pyhf-tutorial#82] [Ideas]
Hi all,

Thanks for putting up such a nice package and the documentations. I wonder whether there has been initiatives to allow pyHF to perform functional fit to data. Since it is a widely adopted analysis methodology, I assume such a new feature will be very welcomed. There are existing HistFactory-based framework that can handle it so it does sound not too crazy. 

Such a feature would also allow us to explore new techniques such as Gaussian Processing Regression and Symbolic Regression easily as those are written in python. 

Any thoughts?

Cheers.

Bing

**Comment by matthewfeickert** [comment]
Hi @prbbing. Sorry this didn't get seen for a long time as it under the tutorial/user guide and not the `pyhf` source repository: https://github.com/scikit-hep/pyhf

Unfortunately, GitHub doesn't allow for transfering GitHub Discussions between GitHub orgs yet (c.f. https://github.com/orgs/community/discussions/5766) so while we can answer this here, can we get you to copy paste this to https://github.com/scikit-hep/pyhf/discussions where more people will find it? I'll follow up and answer this tomorrow (my time).
